### PR TITLE
Add betainc C implementation

### PR DESCRIPTION
**Pull request by arthus701**
## Description
As discussed over at #673, this adds a C-implementation to BetaInc.

## Related Issue

- [x] Related to #673

## Type of change

- [x] New feature / enhancement
- [ ] Bug fix
- [ ] Documentation
- [ ] Maintenance
- [ ] Other (please specify):

### CODE REVIEW

**Comment by ricardoV94** [comment]
Seems like you need to run pre-commit for auto-formatting

**Comment by maresb** [comment]
Could you remove the comma after `self` and reformat? The comma causes the formatter to make a bunch of unnecessary new lines.

**Comment by arthus701** [comment]
Thanks, that also lead me to realize I didn't set a code cache version. I set `0.0.1` now, I hope that's fine.

File: pytensor/scalar/math.py [hunk 1]
```diff
@@ -1495,8 +1495,21 @@ def grad(self, inp, grads):
...
+    def c_code(self, node, name, inp, out, sub):
+        (a, b, x) = inp
+        (z,) = out
+        if node.inputs[0].type in float_types:
+            return f"""{z} = BetaInc({a}, {b}, {x});"""
+
+        raise NotImplementedError("type not supported", type)
+
+    def c_code_cache_version(self):
+        return (0, 0, 1)
```
Comment (ricardoV94) [inline pytensor/scalar/math.py:1512]: Nitpick: return `(1,)`?

**Review by ricardoV94 — COMMENTED** [review]
Looks great, just one nit

**Comment by arthus701** [comment]
Sorry for the delay, I addressed your comments. Thanks for catching the input check.

**Review by ricardoV94 — APPROVED** [review]

**Comment by ricardoV94** [comment]
Sorry for the delay @arthus701 and thanks a lot

**Pull request merged**